magicblock-committor-service/src/persist/db.rs

```diff
@@ -43,166 +45,115 @@ pub struct CommitStatusRow {
     /// For single accounts a bundle_id will be generated as well for consistency
     /// For Pending commits the bundle_id is not set
     pub commit_status: CommitStatus,
+    /// Strategy defined for Commit of a particular account
+    pub commit_strategy: CommitStrategy,
     /// Time since epoch at which the commit was last retried
     pub last_retried_at: u64,
     /// Number of times the commit was retried
     pub retries_count: u16,
 }
 
+#[derive(Debug)]
+pub struct MessageSignatures {
+    /// The signature of the transaction on chain that processed the commit
+    pub processed_signature: Signature,
+    /// The signature of the transaction on chain that finalized the commit
+    /// if applicable
+    pub finalized_signature: Option<Signature>,
+    /// Time since epoch at which the bundle signature was created
+    pub created_at: u64,
+}
+
 impl fmt::Display for CommitStatusRow {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         write!(
             f,
             "CommitStatusRow {{
-    reqid: {}
+    message_id: {}
     pubkey: {},
+    commit_id: {},
     delegated_account_owner: {},
     slot: {},
     ephemeral_blockhash: {},
     undelegate: {},
     lamports: {},
-    finalize: {},
     data.len: {},
     commit_type: {},
     created_at: {},
     commit_status: {},
+    commit_strategy: {},
     last_retried_at: {},
     retries_count: {}
 }}",
-            self.reqid,
+            self.message_id,
             self.pubkey,
+            self.commit_id,
             self.delegated_account_owner,
             self.slot,
             self.ephemeral_blockhash,
             self.undelegate,
             self.lamports,
-            self.finalize,
             self.data.as_ref().map(|x| x.len()).unwrap_or_default(),
             self.commit_type.as_str(),
             self.created_at,
             self.commit_status,
+            self.commit_strategy.as_str(),
             self.last_retried_at,
             self.retries_count
         )
     }
 }
 
-const ALL_COMMIT_STATUS_COLUMNS: &str = r#"
-    reqid,
+const ALL_COMMIT_STATUS_COLUMNS: &str = "
+    message_id,
     pubkey,
+    commit_id,
     delegated_account_owner,
     slot,
     ephemeral_blockhash,
     undelegate,
     lamports,
-    finalize,
-    bundle_id,
     data,
     commit_type,
     created_at,
-    commit_status,
     commit_strategy,
+    commit_status,
     processed_signature,
     finalized_signature,
-    undelegated_signature,
     last_retried_at,
```

Comment:
We need the `undelegated_signature` in case it happened in a separate transaction. If we always undelegate as part of finalize then this is fine.

Reply:
Undelegate doesn't happen per protocol in separate transactions, since there's 2 stages only Commit & Finalize. For now I'm only introducing only optimization for merging them into 1 tx, but undelegate shall follow finalize with actions, otherwise we break atomicity requirments